mitiq/factories.py

```diff
@@ -1,109 +1,164 @@
+"""Contains all the main classes corresponding to different zero-noise extrapolation methods."""
+
 from typing import List, Iterable
 import numpy as np
 
 
-class Factory(object):
+class Factory():
     """
-    This object adaptively produces new noise scaling parameters based on a historical stack of previous noise
-    scale parameters and previously calculated expectation values. Noise scaling parameters are stored on the `in_stack`
-    and the running list of expectation values are stored on the `out_stack`.
+    Abstract class designed to adaptively produce a new noise scaling parameter based on a historical
+    stack of previous noise scale parameters ("self.instack") and previously estimated expectation
+    values ("self.outstack").
+
+    Specific zero-noise extrapolation algorithms, adaptive or non-adaptive, are derived from this class.
+    A Factory object is not supposed to directly perform any quantum computation, only the classical
+    results of quantum experiments are processed by it.
     """
-    def __init__(self, instack: List[float], outstack: List[float]) -> None:
-        self.instack = instack
-        self.outstack = outstack
 
-    def step(self, instack: List[float], outstack: List[float]) -> None:
+    def __init__(self) -> None:
+        """
+        Initialization arguments (e.g. noise scale factors) depend on the particular extrapolation
+        algorithm and can be added to the "__init__" method of the associated derived class.
+        """
+        self.instack = []
```

Comment:
If the user is never supposed to change instack or outstack, they should really be private attributes.

Reply:
e.g., `self._instack = []` and `self._outstack` = []`.

Reply:
The "standard" way updating data is Factory.push(). However I would not completely exclude the the possibility of a direct access to `instack` and `outstack`. For example for transferring the data acquired by a factory into a different factory (as you suggested). Or, more simply, to just visualize the optimization history.  What about if we keep them public for the moment? This is something we can easily change in the future.

Reply:
I think public is good for now. In the future, we should make getter and (maybe) setter methods.